src/components/expression/expression-panel.js

```diff
@@ -224,16 +218,66 @@ genotet.ExpressionPanel.prototype.updateGenes = function(gene) {
       text: gene
     };
   });
-
   var profile = this.container.find('#profile select').empty();
-  this.selectProfiles_ = profile.select2({
-    data: genes,
-    multiple: true
-  });
-  this.container.find('#profile .select2-container').css({
-    width: '100%'
-  });
-  this.updateGeneSelections();
+
+  if (!genes.length) {
+    return;
+  }
+
+  $.fn.select2.amd.require(['select2/data/array', 'select2/utils'],
```

Comment:
A few more comments:

* You may want to rename "genotet.utils.CustomData" to be something more meaningful, such as genotet.utils.Select2PageAdapter.
* Currently the way that select2 works with paging is very hard to use. You can't even search for a gene with known name. More particularly, if the user types in some letters, we need to show only those items that match this search. As far as I know, you can capture what the user types by "params.term" in CustomData.prototype.query. You can then filter the genes by the keywords.
* Another problem is that right now because of the data adapter I don't even see the search box.
![image](https://cloud.githubusercontent.com/assets/1314429/13908308/47fad860-eed6-11e5-8ed4-2a68473561fb.png)
It was something like this before
![image](https://cloud.githubusercontent.com/assets/1314429/13908313/559581c8-eed6-11e5-957a-02c6da608895.png)
so that I can see what I type and search for something.